### ISSUE TITLE
Embed room area as conditioning signal for diffusion model

### ISSUE THREAD

**Issue by nickiliak** [nickiliak/Floorplan-Diffusion-Models#4]
## Description

The diffusion model's conditioning pipeline does not include room area, despite this geometric property being readily available from the source Shapely polygons during data conversion. Adding a normalized area signal would improve geometric plausibility of generated floorplans by letting the model learn size-aware distributions per room type.

## Motivation

Room area is one of the most important geometric properties for realistic floorplan generation. Without it, the model has no direct signal about how large a room should be — it must infer size entirely from coordinate patterns. Embedding area as a conditioning feature would:

- Improve geometric plausibility (e.g., bathrooms should be small, living rooms large)
- Enable area-conditioned generation at inference time (user specifies desired room sizes)
- Require minimal code changes (3 files, ~20 lines)

## Current Behavior

The conditioning pipeline extracts exactly **3 features** per point, totaling **89 dimensions**:

| Feature | Encoding | Dims |
|---------|----------|------|
| `room_type` | One-hot | 25 |
| `corner_index` | One-hot | 32 |
| `room_index` | One-hot | 32 |
| **Total** | | **89** |

These are concatenated and projected through `nn.Linear(89, 512)` in the transformer. Room area is computed nowhere in `src/floorplan_diffusion/` — zero mentions of `area` exist in the package.

## Proposed Change

Add a **single normalized scalar** representing room area as a 90th conditioning dimension.

### Normalization

```python
# In dataset.py, during per-room feature construction
normalized_area = polygon.area / (half_extent ** 2)
```

This keeps area scale-invariant relative to the floorplan's bounding box, consistent with the existing coordinate normalization to `[-1, 1]`.

### Files to Modify

1. **`src/floorplan_diffusion/data/dataset.py`**
   - Compute `polygon.area / half_extent**2` per room during `__getitem__`
   - Add `"room_areas"` key to the conditioning dict (lines 189-198)
   - Broadcast the scalar area to all points belonging to that room

2. **`src/floorplan_diffusion/models/transformer.py`**
   - Add `"room_areas"` to the conditioning loop (lines 425-436)
   - No architectural change needed — the existing concat + linear projection handles it

3. **`scripts/train.py`** and **`scripts/sample.py`**
   - Update `condition_channels` from `89` to `90`

### Alternative: Bucketed Embedding

Instead of a raw scalar, area could be discretized into bins and one-hot encoded (e.g., 8-16 buckets). This is worth exploring if the scalar approach underperforms, but the scalar is simpler and should be tried first.

## Relevant Code

- `src/floorplan_diffusion/data/dataset.py:189-198` — conditioning dict construction
- `src/floorplan_diffusion/models/transformer.py:425-436` — conditioning concatenation and projection
- `scripts/train.py` — `condition_channels = 89` hardcoded
- `scripts/sample.py` — `condition_channels = 89` hardcoded
- `scripts/convert_resplan.py` — source Shapely polygons available here but area not extracted

## Acceptance Criteria

- [ ] `room_areas` key present in dataset conditioning dict
- [ ] Transformer conditions on 90 dims (89 existing + 1 area)
- [ ] Area values are normalized and scale-invariant
- [ ] Existing tests pass with updated `condition_channels`
- [ ] New unit test verifying area is correctly computed and normalized
